runners/direct-java/src/main/java/org/apache/beam/runners/direct/WatermarkManager.java

```diff
@@ -1110,17 +1109,7 @@ public void updateWatermarks(
     pendingUpdates.offer(
         PendingWatermarkUpdate.create(
             executable, completed, timerUpdate, unprocessedInputs, outputs, earliestHold));
-    tryApplyPendingUpdates();
-  }
-
-  private void tryApplyPendingUpdates() {
-    if (refreshLock.tryLock()) {
-      try {
-        applyNUpdates(MAX_INCREMENTAL_UPDATES);
-      } finally {
-        refreshLock.unlock();
-      }
-    }
```

Comment:
Not sure what was the purpose for limiting the number of applied updates, but that seems to be wrong - we need to apply all updates at all times, because otherwise we again might not hold the downstream watermark back enough.